config/routes.rb

```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+   root "users#index"
   # Defines the root path route ("/")
   # root "articles#index"
+  # Users
+  get 'users/:id' => 'users#show'
+
+  # Posts
+  get 'users/:user_id/posts' => 'posts#index'
+  get 'users/:user_id/posts/:id' => 'posts#show'
```

Comment:
- [Optional]  @MohamedCK we could have equally done it using resourceful routing with the **only** constraints to avoid exposing necessary routes.


```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
end
```

Reply:
Hi @FrederickMih, Your suggestion to use resourceful routing is very helpful and appreciated. 😄